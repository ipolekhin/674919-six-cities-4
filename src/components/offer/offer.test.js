import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import Offer from "./offer.jsx";
import NameSpace from "../../reducer/name-space.js";

const authorizationStatus = {
  NO_AUTH: `NO_AUTH`,
};
const placeCards = [
  {
    adults: 3,
    townName: `Amsterdam`,
    bedrooms: 2,
    cardName: `Wood and stone place`,
    cardRating: 4.6,
    cardRatingStars: `92%`,
    cardType: `Room`,
    coordinatesItem: [52.3909553943508, 4.85309666406198],
    description: `Text text....`,
    id: 1,
    image: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    insideItems: [`wi-fi`, `bathroom`],
    premiumPlace: true,
    price: 100,
    reviews: [
      {
        date: new Date(2020, 7, 17),
        id: 1,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        userAvatar: `img/avatar-max.jpg`,
        userName: `Max`,
      }
    ],
    user: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      pro: true,
    },
  },
  {
    adults: 2,
    townName: `Amsterdam`,
    bedrooms: 1,
    cardName: `Wood and stone place`,
    cardRating: 3.6,
    cardRatingStars: `92%`,
    cardType: `Room`,
    coordinatesItem: [52.369553943508, 4.85309666406198],
    description: `Text text....`,
    id: 2,
    image: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    insideItems: [`wi-fi`, `bathroom`],
    premiumPlace: true,
    price: 120,
    reviews: [
      {
        date: new Date(2020, 7, 15),
        id: 1,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        userAvatar: `img/avatar-max.jpg`,
        userName: `John`,
      }
    ],
    user: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      pro: true,
    },
  },
];
const currentCity = `Amsterdam`;
const noop = () => {};
const mockStore = configureStore([]);

describe(`Offer Test`, () => {
  it(`Render Offer`, () => {
    const store = mockStore({
      [NameSpace.REVIEWS]: {
        reviews: [],
      },
    });
    const tree = renderer
      .create(
          <Provider store={store}>
            <Offer
              authorizationStatus = {authorizationStatus.NO_AUTH}
              currentCity = {currentCity}
              onActiveItemChange = {noop}
              placeCards = {placeCards}
            />
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
