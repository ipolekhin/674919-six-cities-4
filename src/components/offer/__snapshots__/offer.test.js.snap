// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Offer 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div>
      <h1>
        3g4g
      </h1>
      <ul>
        <li>
          123
        </li>
        <li>
          223
        </li>
      </ul>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      />
    </div>
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="near-places__list places__list"
      />
    </section>
  </div>
</main>
`;
